common/scripted_effects/wc_pandaria_effects.txt

```diff
@@ -0,0 +1,996 @@
+			change = scope:change
+		}
+	}
+	title:c_huangjia = {
+		holder = {
+			if = {
+				limit = { NOT = { this = scope:reviver } }
+				add_opinion = {
+					target = scope:lei_shen
+					modifier = revoked_title
+					opinion = -30
+				}
+			}
+		}
+		change_title_holder = {
+			holder = scope:lei_shen
+			change = scope:change
+		}
+	}
+	title:c_suzao = {
+		holder = {
+			if = {
+				limit = { NOT = { this = scope:reviver } }
+				add_opinion = {
+					target = scope:lei_shen
+					modifier = revoked_title
+					opinion = -30
+				}
+			}
+		}
+		change_title_holder = {
+			holder = scope:lei_shen
+			change = scope:change
+		}
+	}
+	title:c_duanzao = {
+		holder = {
+			if = {
+				limit = { NOT = { this = scope:reviver } }
+				add_opinion = {
+					target = scope:lei_shen
+					modifier = revoked_title
+					opinion = -30
+				}
+			}
+		}
+		change_title_holder = {
+			holder = scope:lei_shen
+			change = scope:change
+		}
+	}
+	title:c_zhengfu_zin = {
+		holder = {
+			if = {
+				limit = { NOT = { this = scope:reviver } }
+				add_opinion = {
+					target = scope:lei_shen
+					modifier = revoked_title
+					opinion = -30
+				}
+			}
+		}
+		change_title_holder = {
+			holder = scope:lei_shen
+			change = scope:change
+		}
+	}
+	title:c_zhanshi = {
+		holder = {
+			if = {
+				limit = { NOT = { this = scope:reviver } }
+				add_opinion = {
+					target = scope:lei_shen
+					modifier = revoked_title
+					opinion = -30
+				}
+			}
+		}
+		change_title_holder = {
+			holder = scope:lei_shen
+			change = scope:change
+		}
+	}
+	title:b_shandian = {
+		holder = {
+			if = {
+				limit = { NOT = { this = scope:reviver } }
+				add_opinion = {
+					target = scope:lei_shen
+					modifier = revoked_title
+					opinion = -30
+				}
+			}
+		}
+		change_title_holder = {
+			holder = scope:lei_shen
+			change = scope:change
+		}
+	}
+	
+	scope:reviver = {
+		every_held_title = {
+			if = {
+				limit = {
+					tier = tier_empire
+				}
+				change_title_holder = {
+					holder = scope:lei_shen
+					change = scope:change
+				}
+			}
+			else_if = {
+				limit = {
+					OR = {
+						NOT = { exists = scope:secondary_title }
+						tier > scope:secondary_title.tier
+					}
+				}
+				save_temporary_scope_as = secondary_title
+			}
+		}
+	}
+	
+	scope:reviver = {
+		if = {
+			limit = {
+				exists = scope:secondary_title
+			}
+			every_vassal = {
+				limit = {
+					primary_title.tier >= scope:secondary_title.tier
+					NOT = {
+						primary_title.tier = tier_barony
+					}
+				}
+				change_liege = {
+					liege = scope:lei_shen
+					change = scope:change
+				}
+			}
+		}
+	}
+	
+	scope:reviver = {
+		change_liege = {
+			liege = scope:lei_shen
+			change = scope:change
+		}
+	}
+	
+	# If Pandaria isn't his already, get it
+	if = {
+		limit = {
+			NOT = { has_title = title:e_pandaria_empire }
+		}
+		title:e_pandaria_empire = {
+			change_title_holder = {
+				holder = scope:lei_shen
+				change = scope:change
+			}
+		}
+	}
```

Comment:
scenario
background: character 1 establishes the empire of pandaria, destroying their previous empire titles on the continent 
action: character 2 revives the Thunder King 
consequence: character 1 loses their empire title and any kings become independent immediately so that they can decide whether to pledge fealty to Thunder King

just so i understand thats why this is set up to work this way?